vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java

```diff
@@ -0,0 +1,261 @@
+package com.baeldung.vavr.collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Comparator;
+
+import org.junit.Test;
+
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import io.vavr.collection.Array;
+import io.vavr.collection.CharSeq;
+import io.vavr.collection.HashSet;
+import io.vavr.collection.Iterator;
+import io.vavr.collection.List;
+import io.vavr.collection.Map;
+import io.vavr.collection.Queue;
+import io.vavr.collection.Set;
+import io.vavr.collection.SortedMap;
+import io.vavr.collection.SortedSet;
+import io.vavr.collection.Stream;
+import io.vavr.collection.TreeMap;
+import io.vavr.collection.TreeSet;
+import io.vavr.collection.Vector;
+
+public class CollectionAPIUnitTest {
+
+    @Test
+    public void givenEmptyList_whenStacked_thenCorrect() {
+        List<Integer> intList = List.empty();
+
+        List<Integer> anotherList = intList.push(4)
+            .push(0);
+        Iterator<Integer> iterator = anotherList.iterator();
+
+        assertEquals(new Integer(0), iterator.next());
+        assertEquals(new Integer(4), iterator.next());
+    }
+
+    @Test
+    public void givenList_whenPrependTail_thenCorrect() {
+        List<Integer> intList = List.of(1, 2, 3);
+
+        List<Integer> newList = intList.tail()
+            .prepend(0);
+
+        assertEquals(new Integer(1), intList.get(0));
+        assertEquals(new Integer(2), intList.get(1));
+        assertEquals(new Integer(3), intList.get(2));
+
+        assertNotSame(intList.get(0), newList.get(0));
+        assertEquals(new Integer(0), newList.get(0));
+        assertSame(intList.get(1), newList.get(1));
+        assertSame(intList.get(2), newList.get(2));
+    }
+
+    @Test
+    public void givenQueue_whenEnqueued_thenCorrect() {
+        Queue<Integer> queue = Queue.of(1, 2, 3);
+        Queue<Integer> secondQueue = queue.enqueue(4)
+            .enqueue(5);
+
+        assertEquals(3, queue.size());
+        assertEquals(5, secondQueue.size());
+
+        secondQueue.dequeue()
```

Comment:
This is definitely a must-have to show those things but not this way by nesting assertions in _map_ calls and not by nesting _dequeue_ calls inside _map_ calls.

If you want to show that a Tuple gets returned containing _head_ and _tail_, you can start easily with:
`Tuple2<Integer, Queue<Integer>> result = secondQueue.dequeue();`
 and then perform necessary assertions. 

Next thing, if you want to show a _map_ call, definitely do not go with names like _(k,v)_ which suggest that this key-value relationship. Keep in mind that _map_ always needs to return another _Tuple_ 

`result.map((head, tail) -> Tuple.of(head.toString(), tail))`

There is also a method that you can use for transforming the whole tuple into something that is not wrapped in the Tuple:
`result.apply((head, tail) -> head.toString());`

and so on... you can store results of those operations as variables and perform assertions on them.